proof/invariant-abstract/ARM_HYP/ArchArch_AI.thy

```diff
@@ -640,12 +640,11 @@ lemma max_index_upd_no_cap_to:
   apply (clarsimp simp:table_cap_ref_def)
   done
 
-
-lemma perform_asid_control_invocation_st_tcb_at:
-  "\<lbrace>st_tcb_at (P and (Not \<circ> inactive) and (Not \<circ> idle)) t
-    and ct_active and invs and valid_aci aci\<rbrace>
-    perform_asid_control_invocation aci
-  \<lbrace>\<lambda>y. st_tcb_at P t\<rbrace>"
+lemma perform_asid_control_invocation_pred_tcb_at:
+  "\<lbrace>\<lambda>s. pred_tcb_at proj Q t s \<and> st_tcb_at (P and (Not \<circ> inactive) and (Not \<circ> idle)) t s
+        \<and> ct_active s \<and> invs s \<and> valid_aci aci s\<rbrace>
+   perform_asid_control_invocation aci
+   \<lbrace>\<lambda>_. pred_tcb_at proj Q t\<rbrace>"
```

Comment:
Apologies for commenting on this way too late, but I just noticed that the `P` in this precondition no longer seems necessary. Leaving it in like this shouldn't break anything but it might lead to confusing schematic variables when this rule is used

Reply:
Ah yes, you're right, I missed that. We should fix this in a new PR. I don't think it'll break much (or anything), but it's better to not have to worry about unused schematics popping up.